control_apps/src/control_mngr.rs

```diff
@@ -0,0 +1,292 @@
+// ----------------------------------------------------------------------------
+// ControlMngr
+// -----------
+// Top level control application state machine and operations coordinator.
+// ----------------------------------------------------------------------------
+
+use std::fmt;
+use std::thread;
+use std::time::Duration;
+
+use crate::valve::Valve;
+
+const POST_DELAY_MS: u64 = 1000; // bogus delay to emulate POST operations
+const CTRL_ALTITUDE_FLOOR: f32 = 15000.0; // minimum allowed altitude in meters
+const DEFAULT_TARGET_ALTITUDE: f32 = 99999.9; // default target alt in meters
+const CTRL_ERROR_DEADZONE: f32 = 100.0; // magnitude of margin to allow
+
+#[derive(Copy, Clone, PartialEq)]
+pub enum ControlMode {
+    // State machine control mode
+    Init,  // startup, POST, FSW initialization, HW initialization
+    Safe,  // not allowed to actuate valves
+    Idle,  // allowed to actuate valves but currently nothing to do
+    Vent,  // only open descent valve
+    Dump,  // only open ascent valve
+    Abort, // panic! dump all ballast and lock balloon valve closed
+}
+
+pub struct ControlMngr {
+    // Master altitude control state machine
+    mode: ControlMode,
+    valve_vent: Valve,    // valve to actuate in order to lower altitude
+    valve_dump: Valve,    // valve to actuate in order to raise altitude
+    target_altitude: f32, // target altitude hold in meters
+    altitude_error: f32,  // last known altitude error
+}
+
+impl fmt::Display for ControlMode {
```

Comment:
Not sure what the standards are, but might be useful to put this directly after `struct ControlMode`.  I initially thought it was for ControlMngr at first reading.